event.go

```diff
@@ -771,3 +773,22 @@ func (e *Event) MACAddr(key string, ha net.HardwareAddr) *Event {
 	e.buf = enc.AppendMACAddr(enc.AppendKey(e.buf, key), ha)
 	return e
 }
+
+// WithCtx adds a context to the event
+func (e *Event) WithCtx(ctx context.Context) *Event {
```

Comment:
zerolog appears to follow `http`'s naming convention, where this method would be called `WithContext`

Reply:
hmm yeah I wasn't really sure what to do naming wise. Ideally I'd prefer your suggestions (and I think I will change it to them), but I also thought I should differentiate this from [`Logger.WithContext`](https://pkg.go.dev/github.com/rs/zerolog?utm_source=godoc#Logger.WithContext) because that adds a logger to a context (rather than the other way around). So I thought maybe I should follow `Logger.Ctx`](https://pkg.go.dev/github.com/rs/zerolog?utm_source=godoc#Ctx) but that is also working the other way around. Then there's also the confusion added by zerolog having it's own thing called `Context`.